### ISSUE TITLE
Grave stone deletes players inventory

### ISSUE THREAD

**Issue by Allalous** [EnigmaticaModpacks/Enigmatica6#4494]
### Modpack Version

1.0.0 expert

### Describe your issue.

I help run a server and we've had a few players died and either their graves are completely empty or a grave doesn't even spawn. The console doesn't show anything wrong and even says a grave spawns but there's nothing there. We have used the commands to restore their stuff and that works some of the time, but on the occasions that a grave doesn't spawn, we get nothing from there too. There is no crash log either.

### Have you modified the modpack?

No

### Did the issue happen in singleplayer or on a server?

Server

**Comment by FooterMan14** [comment]
I'll just add a thing here, you should make sure if you have some sort of lag removal tool to blacklist graves from being cleared as its an entity

**Comment by Allalous** [comment]
I'll check, but I don't think we have anything like that

**Comment by Saereth** [comment]
Several people have reported this on the Breakinblocks server as well. We have no autopruning mods, strictly stock forge and the pack, no other changes. As it has occurred the graves have completely failed to generate a file. We cannot reliably reproduce it but it has happened to several players already and is such an inconvenience we're considering putting on keepinventory at this point which is far from ideal.

**Comment by NillerMedDild** [comment]
Crap :/ thanks for weighing in Saereth. If you find a way to reproduce it, please let us know asap :D

**Comment by NillerMedDild** [comment]
@Quarris halp 😅

**Comment by Allalous** [comment]
I had a talk with the other admins on the server and yeah we have no lag removal tool and we even tried to get other players to die over and over again to get something to show up, but either it wouldn't work right (as is the mod was working right with getting their grave and stuff back) or one time they had 2 grave spawn from other death. That one was odd because the console only showed one being generated not 2, but luckily it didn't dupe anything.

**Comment by Quarris** [comment]
I just looked and the debug logs might only store like the recent 5 logs :/ It would be in the `logs` folder in the root modpack directory. If in doubt, send all `debug-#.log.gz` folders, I'll see if any of them are of help

**Comment by Allalous** [comment]
[debug-1.log.gz](https://github.com/EnigmaticaModpacks/Enigmatica6/files/8373945/debug-1.log.gz)
[debug-2.log.gz](https://github.com/EnigmaticaModpacks/Enigmatica6/files/8373947/debug-2.log.gz)
[debug-3.log.gz](https://github.com/EnigmaticaModpacks/Enigmatica6/files/8373948/debug-3.log.gz)
[debug-4.log.gz](https://github.com/EnigmaticaModpacks/Enigmatica6/files/8373949/debug-4.log.gz)
[debug-5.log.gz](https://github.com/EnigmaticaModpacks/Enigmatica6/files/8373950/debug-5.log.gz)

**Comment by Allalous** [comment]
if they don't show anything, I'll ask the players to die again for science haha

**Comment by Quarris** [comment]
These debugs dont show anything, but I worry that maybe the debug logging doesnt actually log past a certain point? Because all these logs end after the world has loaded, and then just dont show anything after that. Maybe youre able to get the normal logs at the appropriate date, they might tell me something different

**Comment by Quarris** [comment]
Actually the debug logs should be enough, maybe the server restarted a few times without anything happening which is why the logs look so empty :/

**Comment by Quarris** [comment]
Im moving this issue into the mod issue tracker. I'll explain more there

**Comment by Quarris** [comment]
If you could go to the bug report mention above. I released an `ALPHA` version for 1.7.4 which adds a lot more logging info. Im not sure whether the server youre using for testing is Enigmatica or a custom pack; but if you're able to do some tests with the alpha version on CF, and immediately get the debug logs when it happens, that'd be amazing.

@NillerMedDild Not sure if you want to update the back with the new version, it changes nothing other than all the logs, but hopefully it'll help track down the bugs that people are having because currently I'm really struggling with these.

**Comment by Allalous** [comment]
I will let you know what happens, thanks!! keep up the good work everyone!

**Comment by TheDragonicSlayer** [comment]
I don't think the grave deletes your inventory. It just forces the items out of the inventory and drops them and the items in the grave are forced back to their slots. Then you pickup the items. This feature is problematic if you are in lava so when the items are dropped they are destroyed

**Comment by Allalous** [comment]
When we were running our tests, we never saw items dropping on the ground when a grave was made.

**Comment by Quarris** [comment]
> I don't think the grave deletes your inventory. It just forces the items out of the inventory and drops them and the items in the grave are forced back to their slots. Then you pickup the items. This feature is problematic if you are in lava so when the items are dropped they are destroyed

The only time when items should directly drop on the ground, is if you don't have the space to hold your old inventory and you current inventory

**Comment by User-12691296** [comment]
Same thing just happened to my friend, the grave wasn't gone, it just placed it at the exact same coords in a different dimension.

**Comment by tyler489** [comment]
> > I don't think the grave deletes your inventory. It just forces the items out of the inventory and drops them and the items in the grave are forced back to their slots. Then you pickup the items. This feature is problematic if you are in lava so when the items are dropped they are destroyed
> 
> The only time when items should directly drop on the ground, is if you don't have the space to hold your old inventory and you current inventory

This is incorrect in my experiences, if you have something in that slot EG like a set of armor, bursting a grave sends the second set of armor (not in the grave) Flying, regardless of if you have space or not, happens for every slot.

**Comment by Quarris** [comment]
> > > I don't think the grave deletes your inventory. It just forces the items out of the inventory and drops them and the items in the grave are forced back to their slots. Then you pickup the items. This feature is problematic if you are in lava so when the items are dropped they are destroyed
> > 
> > 
> > The only time when items should directly drop on the ground, is if you don't have the space to hold your old inventory and you current inventory
> 
> This is incorrect in my experiences, if you have something in that slot EG like a set of armor, bursting a grave sends the second set of armor (not in the grave) Flying, regardless of if you have space or not, happens for every slot.

I have indeed managed to replicate this and believe have fixed the issue in 1.7.4-hf2
